### PR TITLE
[Fixes #5352] When replacing Controller.Dispose with an explicit impl…

### LINKED ISSUE #5352: When replacing Controller.Dispose with an explicit implementation the base Dispose is an action
```
public class HomeController : Controller, IDisposable
{
        void IDisposable.Dispose()
        {
            Debug.WriteLine("Got disposed");
        }
}
```

If you define this controller, then `/Home/Dispose` will be routed to the `Controller.Dispose()` method and will call it as an action.

Note that this requires you to reimplement `IDisposable` on the subclass. Just overriding the `protected void Dispose()` would not have any deleterious impact.

This is a similar case that also reproduces the bug:

```
public class HomeController : Controller
{
        public new void Dispose()
        {
            Debug.WriteLine("Got disposed");
        }
}
```

### PR DESCRIPTION
**Pull request by kichalla**
…ementation the base Dispose is an action

### CODE REVIEW

**Comment by ryanbrandenburg** [comment]
Looks good to me.

File: test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/DefaultApplicationModelProviderTest.cs [hunk 1]
```diff
@@ -883,6 +883,41 @@ public void CreateActionModel_InheritedAttributeRoutesOverridden()
...
+            // Act
+            builder.OnProvidersExecuting(context);
+
+            // Assert
+            var model = Assert.Single(context.Result.Controllers);
+            Assert.Empty(model.Actions);
+        }
+
+        [Fact]
+        public void ControllerDispose_NotAnAction_WhenDerivedController_HidesControllerDispose()
```
Comment (pranavkm) [inline test/Microsoft.AspNetCore.Mvc.Core.Test/Internal/DefaultApplicationModelProviderTest.cs]: `MethodsNamedDisposeAreTreatedAsActions`

**Comment by rynowak** [comment]
Sorry for not commenting as well, I discussed this in person with @kichalla - the intent here is that an `IDisposable` implementation will **never** be considered an action. We have code like this already, but it has a bug. You shouldn't need to put `[NonAction]` on `Dispose`.

**Comment by kichalla** [comment]
🆙 📅 Folks, please take a look again.

**Pull request merged**